letter-shop.md

```diff
@@ -0,0 +1,232 @@
+
+## Paying for your letter
+Visit http://localhost:8000. You'll see something like:
+```txt
+Please send an Interledger payment to test.crypto.xrp.rrhnXcox5bEmZfJCHzPxajUtwdt772zrCW with condition A6-zI1uIEtjOXMTDoZLtML1xj6YPxBA6yxIQyVh4qhE
+```
+
+Save the following script as `pay.js`:
+```js
+const Plugin = require('ilp-plugin-xrp-escrow')
+const uuid = require('uuid/v4')
+
+const plugin = new Plugin({
+  secret: 'sndb5JDdyWiHZia9zv44zSr2itRy1',
+  account: 'rGtqDAJNTDMLaNNfq1RVYgPT8onFMj19Aj',
+  server: 'wss://s.altnet.rippletest.net:51233',
+  prefix: 'test.crypto.xrp.'
+})
+
+function sendTransfer (obj) {
+  obj.id = uuid()
+  obj.from = plugin.getAccount()
+  // to
+  obj.ledger = plugin.getInfo().prefix
+  // amount
+  obj.ilp = 'AA'
+  obj.noteToSelf = {}
+  // executionCondition
+  obj.expiresAt = new Date(new Date().getTime() + 1000000).toISOString()
+  obj.custom = {}
+  return plugin.sendTransfer(obj)
+}
+
+plugin.connect().then(function () {
+  plugin.on('outgoing_fulfill', function (transferId, fulfillment) {
+    console.log('Got the fulfillment, you paid for your letter! Go get it at http://localhost:8000/' + fulfillment)
+    plugin.disconnect()
+    process.exit()
+  })
+
+  sendTransfer({
+    to: process.argv[2],
+    amount: '1',
+    executionCondition: process.argv[3]
+  }).then(function () {
+    console.log('transfer prepared, waiting for fulfillment...')
+  }, function (err) {
+    console.error(err.message)
+  })
+})
+```
+
+Now run, something like the following, (put the condition from your own shop's local website as the second argument):
+```sh
+$ node ./pay.js test.crypto.xrp.rrhnXcox5bEmZfJCHzPxajUtwdt772zrCW A6-zI1uIEtjOXMTDoZLtML1xj6YPxBA6yxIQyVh4qhE
+```
+
+Now wait for about 30 seconds, until you see something like:
+```txt
+Got the fulfillment, you paid for your letter! Go get it at http://localhost:8000/RWtoGF_sOVoIH3Casd-nmApQ-Thzl03lH-cInXume_g
+```
+
+As the instructions say, visit that URL to get your letter! :)
+
+## Paying proxy
+
+It's of course very cumbersome to cut and paste the condition from your browser to your command-line terminal each time you need to pay for something online, and then to cut and past back the fulfillment from your terminal to your browser once you paid. Therefore, the following paying proxy is useful, which parses the shop's payment instructions, executes them, retreives the paid content, and serves it on port 8001. Save it as `proxy.js`:
+```js
+const Plugin = require('ilp-plugin-xrp-escrow')
+const http = require('http')
+const fetch = require('node-fetch')
+const uuid = require('uuid/v4')
+
+const plugin = new Plugin({
+  secret: 'sndb5JDdyWiHZia9zv44zSr2itRy1',
+  account: 'rGtqDAJNTDMLaNNfq1RVYgPT8onFMj19Aj',
+  server: 'wss://s.altnet.rippletest.net:51233',
+  prefix: 'test.crypto.xrp.'
+})
+
+const pendingRes = {}
+
+function sendTransfer (obj) {
+  obj.id = uuid()
+  obj.from = plugin.getAccount()
+  // to
+  obj.ledger = plugin.getInfo().prefix
+  // amount
```

Comment:
Wait, where's the amount?

Reply:
All this function does is add the "useless" fields. You still need to set the "required" fields yourself, see for instance how it gets called [here](https://github.com/interledger/tutorials/blob/dcde0af71854fc15c38a209a53f43263967287db/pay.js#L32-L36).